src/common/components/PlayForms.jsx

```diff
@@ -1,91 +1,123 @@
-                                      value={formData[field.datafields]} 
-                                      size="small"
-                                      className='w-full'
-                                      {...field}
+  const [formData, setFormData] = useState({...formDataObj});
 
-                                      onChange={(e) => {
-                                        onDataChanged(field.datafield, e.target.value)}
-                                      }/>
-        case 'select': return <Autocomplete id={field.datafield}
-                                            size="small"
-                                            options={field.options || []}
-                                            getOptionLabel={(option) => option.name || option[field.fieldName] || option}
-                                            filterSelectedOptions
-                                            multiple={field.multiple}
-                                            freeSolo={field.freeSolo}
-                                            onChange={(e, newValue) => {
-                                              let updatedval = newValue;
-                                              if(field.multiple) {
-                                                updatedval = [];
-                                                newValue.forEach(v => {
-                                                  if(_.isObject(v)){
-                                                    updatedval.push(v);
-                                                  } else {
-                                                    updatedval.push({
-                                                      [field.fieldName || 'name']: v,
-                                                      [field.fieldValue || 'value']: ""
-                                                    })
-                                                  }
-                                                })
-                                              }
-                                              onDataChanged(field.datafield, updatedval)}
-                                            }
-                                            renderInput={(params) => (
-                                              <TextField
-                                                {...params}
-                                                size="small"
-                                                placeholder={field.placeholder}
-                                              />
-                                            )}
-                                />
-                default: return <></>
-    }
-  }
+  const checkDisabledInputs = (fieldName) => isEditPlay && disabledFields.includes(fieldName)
+
+  const handleChange = (key, value) => {
+    setFormData((pre) => ({...pre, [key]: value }));
+  };
+
+  useEffect(() => {
+    setFormData({...formDataObj})
```

Comment:
```suggestion
    setFormData(ormDataObj)
```

Reply:
here resetting the state to the default cannot set formDataObj again our main state.